java-engine/src/main/java/io/getunleash/engine/YggdrasilFFI.java

```diff
@@ -69,4 +82,20 @@ Pointer checkEnabled(String name, String context){
     Pointer checkVariant(String name, String context){
         return this.ffi.check_variant(this.enginePtr, name, context);
     }
+
+    private static final class YggdrasilNativeLibraryResourceCleaner implements Runnable {
```

Comment:
A whole Runnable for this?

Java, you've been drinking again, haven't you?

Jokes aside, this looks sane

Reply:
Yeah, also because of:
> The cleaning action could be a lambda but all too easily will capture the object reference, by referring to fields of the object being cleaned, preventing the object from becoming phantom reachable. Using a static nested class, as above, will avoid accidentally retaining the object reference.